### PR TITLE
feat: pass e2e unleash experiments as a header to Metaphysics

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Feat**

### Description

This PR is a proof-of-concept and one of a few moving parts for the "[Share split-test assignments with back-ends, for safer data/API roll-outs](https://www.notion.so/artsy/Share-split-test-assignments-with-back-ends-for-safer-data-API-roll-outs-7dcfb3401f1d4fb3b26af6dc70a3315b#42f1b1213b8f4ffaa89258ab0444840a)" technical plan.

The idea is that Force and Eigen detect e2e experiments (by "e2e_" prefix in their name) and sends such experiments in "X-Variants" header to Metaphysics.

For demo and testing purposes, I've created two Unleash experiments, they are enabled for dev/staging:

- [One](https://unleash.artsy.net/projects/default/features/e2e_test_nikita)
- [Two](https://unleash.artsy.net/projects/default/features/e2e_test_2_nikita)

You can test it in action in the [review app](https://e2e-unleash-experiments.artsy.net/).

### CODE REVIEW

**Comment by nickskalkin** [comment]
Interesting, I do not see "X-Variants" header until I log in

File: src/Server/middleware/featureFlagMiddleware.ts [hunk 1]
```diff
@@ -43,7 +43,14 @@ export function featureFlagMiddleware(serviceType: symbol) {
           }
         }, {})
 
+        const e2eFeatureFlags = Object.fromEntries(
+          Object.entries(featureFlags).filter(
+            ([key, value]) => key.startsWith("e2e_") && value.flagEnabled
+          )
+        )
```
Comment (joeyAghion) [inline src/Server/middleware/featureFlagMiddleware.ts:50]: Despite what we proposed in the tech plan, [this recommends hyphenating feature names](https://www.notion.so/artsy/Feature-toggles-and-experiments-43523493b48a4a92882037f0106bef5c?pvs=4#258f640ba019472db81d3c114fd03132) so maybe we should align with that.

**Review by joeyAghion — APPROVED** [review]
I'm pleasantly surprised by how little code this requires! Will try the review app...

**Comment by joeyAghion** [comment]
Logged out, I saw a partial `X-Variants` header being attached to Metaphysics requests associated with the initial page-load:
```
X-Variants: {"e2e_test_2_nikita":{"flagEnabled":true,"variant":{"name":"disabled","enabled":false}}}
```

But I didn't see the header added to later requests when I navigated around (client-side). After logging in, I seemed to see the full list on all requests:

```
X-Variants: {"e2e_test_nikita":{"flagEnabled":true,"variant":{"name":"disabled","enabled":false}},"e2e_test_2_nikita":{"flagEnabled":true,"variant":{"name":"disabled","enabled":false}}}
```

**Comment by mzikherman** [comment]
Sweet!

**Pull request closed without merging**